modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -4205,12 +4123,43 @@ private void executeApplicationFlow(String flowName) throws MuleException {
         .build());
   }
 
-  private void assertZombieApplication(String appId) {
-    final Map<URI, Long> zombieMap = deploymentService.getZombieApplications();
-    assertEquals("Wrong number of zombie apps registered.", 1, zombieMap.size());
-    final Map.Entry<URI, Long> zombie = zombieMap.entrySet().iterator().next();
-    assertEquals("Wrong URL tagged as zombie.", appId, new File(zombie.getKey()).getParentFile().getName());
-    assertTrue("Invalid lastModified value for file URL.", zombie.getValue() != -1);
+  private void assertZombieApplication(String appId, int expectedZombieCount, boolean parent) {
+    final Map<String, Map<URI, Long>> zombieMap = deploymentService.getZombieApplications();
+    assertZombie(zombieMap, appId, expectedZombieCount, parent);
+  }
+
+  private void assertZombieDomain(String domainId, int expectedZombieCount, boolean parent, boolean expectName) {
+    final Map<String, Map<URI, Long>> zombieMap = deploymentService.getZombieDomains();
+    assertZombie(zombieMap, domainId, expectedZombieCount, parent);
+  }
+
+  private void assertZombie(Map<String, Map<URI, Long>> zombieMap, String appId, int expectedZombieCount, boolean parent) {
```

Comment:
Is not clear what the "parent" parameter does. Can you split the method and make it more clear in the name which are the differences usages?